py/server/deephaven/rollup_table.py

```diff
@@ -0,0 +1,195 @@
+
+    def node_operation_recorder(self, node_type: NodeType) -> RollupNodeOperationsRecorder:
+        """Creates a RollupNodeOperationsRecorder for per-node operations to apply during snapshotting of this
+        RollupTable. The recorded node operations will be applied only to the node of the provided NodeType. See
+        :class:`NodeType` for details.
+
+        Args:
+            node_type (NodeType): the type of node tables that the recorded operations will be applied to; if it is
+                :attr:`NodeType.CONSTITUENT`, the RollupTable must be created with include_constituents=True.
+
+        Returns:
+            a RollupNodeOperationsRecorder
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupNodeOperationsRecorder(j_node_ops_recorder=self.j_rollup_table.makeNodeOperationsRecorder(
+                node_type.value))
+        except Exception as e:
+            raise DHError(e, "failed to create a RollupNodeOperationsRecorder.") from e
+
+    def with_node_operations(self, node_ops_recorders: List[RollupNodeOperationsRecorder]) -> RollupTable:
+        """Returns a new RollupTable that will apply the recorded node operations to nodes when gathering
+        snapshots.
+        
+        Args:
+            node_ops_recorders (List[RollupNodeOperationsRecorder]): a list of RollupNodeOperationsRecorder containing
+                the node operations to be applied
+
+        Returns:
+            a new RollupTable
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupTable(
+                j_rollup_table=self.j_rollup_table.withNodeOperations(
+                    [op.j_node_ops_recorder for op in node_ops_recorders]),
+                include_constituents=self.include_constituents, aggs=self.aggs, by=self.by)
+        except Exception as e:
+            raise DHError(e, "with_node_operations on RollupTable failed.") from e
+
+    def with_filters(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]) -> RollupTable:
+        """Applies a set of filters to the group-by columns of this RollupTable to produce and return a new
+        RollupTable
+
+        Args:
+            filters (Union[str, Filter, Sequence[str], Sequence[Filter]], optional): the filter condition
+                expression(s) or Filter object(s)
+
+        Returns:
+            a new RollupTable
+
+        Raises:
+            DHError
+        """
+        try:
+            filters = to_sequence(filters)
+            if filters and isinstance(filters[0], str):
+                filters = Filter.from_(filters)
+            filters = j_array_list(filters)
+
+            return RollupTable(j_rollup_table=self.j_rollup_table.withFilters(filters),
+                               include_constituents=self.include_constituents, aggs=self.aggs, by=self.by)
+        except Exception as e:
+            raise DHError(e, "with_filters operation on RollupTable failed.") from e
```

Comment:
`RollupNodeOperationsRecorder` appears to be the only outwardly facing recorder in the `RollupTable` API.  Does it make any sense to make the basic recorders private?